src/app/admin/websiteDataTable/CustomEditComponent.tsx

```diff
@@ -0,0 +1,57 @@
+import { GridRenderEditCellParams, useGridApiContext } from "@mui/x-data-grid";
+import React, { useEffect, useLayoutEffect, useRef, useState } from "react";
+
+interface CustomCellProps extends GridRenderEditCellParams {
+    editMode: boolean;
+}
+
+const CustomComponent: React.FC<CustomCellProps> = (props: CustomCellProps) => {
```

Comment:
inconsistency between filename and the component name: though I'm not sure what CustomEditComponent would imply. Maybe `EditableTextAreaForDataGrid` or something? (since this component uses useGridApiContext which requires it to be inside the data grid)

Reply:
Yep sure. I was lazy here